app/lib/tool/test_profile/import_source.dart

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:archive/archive.dart';
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+
+import '../../shared/urls.dart' as urls;
+
+import 'models.dart';
+import 'resolver.dart' as resolver;
+
+/// Interface for resolving and getting data for profiles.
+abstract class ImportSource {
+  /// Resolve all the package-version required for the [profile].
+  Future<List<ResolvedVersion>> resolveVersions(TestProfile profile);
+
+  /// Gets the archive bytes for [package]-[version].
+  Future<List<int>> getArchiveBytes(String package, String version);
+
+  /// Close resources that were opened during the sourcing of data.
+  Future<void> close();
+
+  /// Creates a source that resolves and downloads data from pub.dev.
+  static ImportSource fromPubDev({
+    @required String archiveCachePath,
+  }) {
+    return _PubDevImportSource(archiveCachePath: archiveCachePath);
+  }
+
+  /// Creates a source that generates data based on random seed, without any
+  /// network (or file) access.
+  static ImportSource semiRandom() {
```

Comment:
Wouldn't it be nice to all the `pseudoRandom` or `arbitrary` or  `generatedTestData` or `autoGenerated` or...